### PR TITLE
Use utop as a library for rtop

### PR DESCRIPTION
**Pull request by ulrikstrid**
Use `UTop_main.main ()` to setup the utop toplevel to reduce complexity of implementation

### CODE REVIEW

File: src/rtop/rtop.ml [hunk 1]
```diff
@@ -1,32 +1,25 @@
...
-let home = match getenv_opt "HOME" with
-  | Some(p) -> p
-  | None -> Sys.getenv "USERPROFILE"
+let () = try Topdirs.dir_directory (Sys.getenv "OCAML_TOPLEVEL_PATH") with | Not_found -> ();;
 
-let () = Unix.openfile (Filename.concat home ".utoprc") [] 0o640
-         |> Unix.close
+let () = UTop.require ["reason.easy_format"; "reason";]
 
-let () = Unix.openfile (Filename.concat home ".utop-history") [] 0o640
```
Comment (jordwalke) [inline src/rtop/rtop.ml]: I wonder if it now wouldn't preserve history or load settings.

**Comment by ulrikstrid** [comment]
The diff is random, it probably makes more sense to compare `rtop.ml` with `rtop_init.ml` as that is what I reimplemented

File: src/rtop/rtop.ml (hunk 1 above)
Comment (ulrikstrid) [reply src/rtop/rtop.ml]: It kept history when I tested while building it using `esy x rtop`, my guess is that it just loads `.utop-history` or something like that anyway.

Comment (hcarty) [reply src/rtop/rtop.ml]: If it loads the same history as utop won't that cause issues on systems where both utop and rtop are used?

Comment (ulrikstrid) [reply src/rtop/rtop.ml]: I didn't get any history in rtop after using utop but I might have to test this some more

Comment (ulrikstrid) [reply src/rtop/rtop.ml]: Just tried again, can confirm that they do not share history. But it might be because esy sandboxes my rtop.

Comment (jordwalke) [reply src/rtop/rtop.ml]: We should make sure it doesn't fail on a clean install or at least figure out where it's getting the history from - it's getting it from *somewhere*. This was an original bug we had (the history file didn't exist) and rtop failed on first launch.

Comment (ulrikstrid) [reply src/rtop/rtop.ml]: https://github.com/facebook/reason/blob/master/src/rtop/reason_utop.ml#L54

This is where it comes from

Comment (hcarty) [reply src/rtop/rtop.ml]: That's promising - it should continue to use an rtop-specific history file then.

**Pull request merged**